Classes/MDMPersistenceController/MDMPersistenceController.m

```diff
@@ -258,6 +275,57 @@ - (NSManagedObjectContext *)newChildManagedObjectContext {
     return childManagedObjectContext;
 }
 
+#pragma mark - Independent Context 
+
+- (NSManagedObjectContext *)createPrivateManagedObjectContextWithNewPersistentStoreCoordinator {
+    //Based on https://github.com/mmorey/MDMHPCoreData
+    
+    if (self.managedObjectContext == nil) {
+        // The primary persistent stack should have been initialized as part of this object's initialization - did it fail?
+        ALog(@"WARNING: Main context should have already been initialized by now!");
+        //return nil;
+    }
+    
+    // Setup new persistent store coordinator
+    NSPersistentStoreCoordinator *persistentStoreCoordinator = [self setupNewPersistentStoreCoordinator];
+    if (persistentStoreCoordinator == nil) {
+        return nil;
+    }
+    
+    // Create private managed object context
+    NSManagedObjectContext *privateContext = [[NSManagedObjectContext alloc] initWithConcurrencyType:NSPrivateQueueConcurrencyType];
+    [privateContext setPersistentStoreCoordinator:persistentStoreCoordinator];
+    if (privateContext == nil) {
+        ALog(@"ERROR: Failed to create managed object context");
+        return nil;
+    }
+    
+    // Setup observer to receive this context's save operation completion and further broadcast using predefined notification name.
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(independentManagedObjectContextDidSaveNotification:)
+                                                 name:NSManagedObjectContextDidSaveNotification
+                                               object:(privateContext)];
```

Comment:
@ParmeshwarBayappu I was actually talking about the `object:(privateContext)` parenthesis.